src/components/ui/Input/SquareTextInput.tsx

```diff
@@ -0,0 +1,99 @@
+'use client';
+import { CircleCrossIcon } from '@/assets/images/icons';
+import { cn } from '@/lib/tailwind/utils';
+
+import { VariantProps } from 'class-variance-authority';
+import { forwardRef, InputHTMLAttributes, useEffect, useRef } from 'react';
+
+import { Button } from '../Button';
+import { Icon, iconVariants } from '../Icon';
+import { Input, inputVariants } from './Input';
+
+interface ISquareTextInputProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> {
+  /**
+   * Executed when the cross button is clicked
+   */
+  handleCrossButtonClick: () => void;
+  /**
+   * Executed when outside of the input is clicked. Will not be invoked when cross button is clicked.
+   */
+  handleOutsideClick?: () => void;
```

Comment:
handleOutsideClick is optional in here, since I wasn't sure if it's needed in everywhere this component is used.